layers/+lang/purescript/funcs.el

```diff
@@ -37,3 +37,46 @@
      (spacemacs|add-company-backends
        :backends company-psc-ide-backend
        :modes purescript-mode))))
+
+(defun spacemacs/purescript-format ()
+  "Call formatting tool specified in `purescript-fmt-tool'."
+  (interactive)
+  (call-interactively
+   (pcase purescript-fmt-tool
+     ('purs-tidy 'spacemacs/purescript-purs-tidy-format-buffer)
+     (_ (user-error
+         "%s isn't a valid purescript formatter. Possible values are 'purs-tidy"
+         purescript-fmt-tool)))))
+
+(defun spacemacs/purescript-purs-tidy-format-buffer ()
+  "Format buffer with purs-tidy."
+  (interactive)
+  (if (executable-find "purs-tidy")
+      (let*  ((extension (file-name-extension (or buffer-file-name "tmp.purs") t))
+              (tmpfile (make-temp-file "~fmt-tmp" nil extension))
+              (coding-system-for-read 'utf-8)
+              (coding-system-for-write 'utf-8)
+              (outputbuf (get-buffer-create "*~fmt-tmp.purs*")))
+        (unwind-protect
```

Comment:
For this `unwind-protect` form:

First, there's no need for `progn` in for the ELSE part in a `if` form:
```elisp
(progn
  (message "Formatting failed!")
  (display-buffer outputbuf)))
```

Second, it seems that
```elisp
(progn
  (delete-file tmpfil)))))
```
is intend to be the clean up forms should the body form `unwind-protect` fails. But it's now in the body form, due to the missing parenthesis after `(display-buffer outputbuf)))`.

Reply:
You are right about the unwind-protect, the temporary file should always be deleted on cleanup. Fixed now.

Regarding the progn, I find it clearer to group the statements for else together, however I'm okay either way. I have removed the extra progn.